File: python/7_kyu/homogeneous_arrays.py
# Homogenous arrays
# https://www.codewars.com/kata/57ef016a7b45ef647a00002d/train/python

def filter_homogenous(arrays):
  ans = []
  
  for i in arrays:
    if len(i) >= 1:
      char_type = type(i[0])
    
      for j in i:
        match = True
        if type(j) != char_type:
          match = False
          break
    
      if match:
        ans.append(i)
  
  return ans
    


print(filter_homogenous([[935, 768, 661, 'mhas', 'eopa', 945, 381], ['ylkn', 'eauj', 'msdj', 'dqrq', 'psag', 'rynt', 'lebs'], [358, 239, 165, 724, 820], [479, 673, 61, 347, 211, 552, 882, 879], [451, 'bnyg', 116, '', 560, 214, 599], [777, 'abxt', 921, 96, 958, 647], ['efmm', 672, 472, '', 'dalc'], ['rpel', 586, 'obxh', 336, 330, 93, 'fdov'], [663, 534, '', 615], ['xvsl', 'kfzv'], ['yjif', 'vaid', 'qjju', '', 'twlz', 'ywxa', 'patp', 'xdrf', 'lbng'], [244, 993, 966, 338, 876, 904, 324]]))

File: javascript/7_kyu/homogeneous_arrays.js
// Homogenous arrays
// https://www.codewars.com/kata/57ef016a7b45ef647a00002d/train/javascript

function filterHomogenous(arrays) {
  var ans = [];

  for (var i = 0; i < arrays.length; i++) {
    if ( arrays[i].length >= 1 ) {
      var char_type = typeof arrays[i][0];

      for (var j = 0; j < arrays[i].length; j++) {
        var match = true;
        if ( !(typeof arrays[i][j] == char_type) ) {
          match = false;
          break;
        }
      }
      if (match == true) {
        ans.push(arrays[i]);
      }
    }
  }
  return ans;
}

console.log(filterHomogenous([[1, 5, 4], ['a', 3, 5], ['b'], [], ['1', 2, 3]]));

// // Best Solution
// let filterHomogenous = a => a.filter(b => b.length > 0 && b.every(e => typeof e == typeof b[0]));